text/0000-javascript-module-api.md

```diff
@@ -0,0 +1,1058 @@
+| ---                                             | ---                      |
+| `import { run } from "@ember/runloop"`          | `Ember.run`              |
+| `import { begin } from "@ember/runloop"`        | `Ember.run.begin`        |
+| `import { bind } from "@ember/runloop"`         | `Ember.run.bind`         |
+| `import { cancel } from "@ember/runloop"`       | `Ember.run.cancel`       |
+| `import { debounce } from "@ember/runloop"`     | `Ember.run.debounce`     |
+| `import { end } from "@ember/runloop"`          | `Ember.run.end`          |
+| `import { join } from "@ember/runloop"`         | `Ember.run.join`         |
+| `import { later } from "@ember/runloop"`        | `Ember.run.later`        |
+| `import { next } from "@ember/runloop"`         | `Ember.run.next`         |
+| `import { once } from "@ember/runloop"`         | `Ember.run.once`         |
+| `import { schedule } from "@ember/runloop"`     | `Ember.run.schedule`     |
+| `import { scheduleOnce } from "@ember/runloop"` | `Ember.run.scheduleOnce` |
+| `import { throttle } from "@ember/runloop"`     | `Ember.run.throttle`     |
+
+### `@ember/service`
+| Module                                    | Global                 |
+| ---                                       | ---                    |
+| `import Service from "@ember/service"`    | `Ember.Service`        |
+| `import { inject } from "@ember/service"` | `Ember.inject.service` |
+
+### `@ember/string`
+| Module                                       | Global                    |
+| ---                                          | ---                       |
+| `import { camelize } from "@ember/string"`   | `Ember.String.camelize`   |
+| `import { capitalize } from "@ember/string"` | `Ember.String.capitalize` |
+| `import { classify } from "@ember/string"`   | `Ember.String.classify`   |
+| `import { dasherize } from "@ember/string"`  | `Ember.String.dasherize`  |
+| `import { decamelize } from "@ember/string"` | `Ember.String.decamelize` |
+| `import { fmt } from "@ember/string"`        | `Ember.String.fmt`        |
+| `import { htmlSafe } from "@ember/string"`   | `Ember.String.htmlSafe`   |
+| `import { loc } from "@ember/string"`        | `Ember.String.loc`        |
+| `import { underscore } from "@ember/string"` | `Ember.String.underscore` |
+| `import { w } from "@ember/string"`          | `Ember.String.w`          |
+
+### `@ember/utils`
+| Module                                     | Global            |
+| ---                                        | ---               |
+| `import { isBlank } from "@ember/utils"`   | `Ember.isBlank`   |
+| `import { isEmpty } from "@ember/utils"`   | `Ember.isEmpty`   |
+| `import { isNone } from "@ember/utils"`    | `Ember.isNone`    |
+| `import { isPresent } from "@ember/utils"` | `Ember.isPresent` |
+| `import { tryInvoke } from "@ember/utils"` | `Ember.tryInvoke` |
+| `import { typeOf } from "@ember/utils"`    | `Ember.typeOf`    |
+
```

Comment:
It is likely that `Ember.K` will be deprecated in some future version (likely after an RFC to explain all the reasoning behind the deprecation of an oft-used method).

Consider:

```js
{
  foo: Ember.K
}
```

vs:

```js
{
  foo(){}
}
```

The latter is shorter, and less "magic".  

Most people are pretty surprised at the chaining aspect of `Ember.K`:

```js
let derp = {
  foo: Ember.K,
  bar: Ember.K,
  baz: Ember.K
}

derp.foo().bar().baz();

// O_o
```

Another large downside is that if the same function is used in many places, and that function is deopted then all of the locations it is used in are affected. Obviously, this sucks 😛